File: 1646-KthMissingPositiveNumber/1646-KthMissingPositiveNumber.py
# Last updated: 1/26/2026, 8:38:43 AM
class Solution(object):
    def findKthPositive(self, arr, k):
        """
        :type arr: List[int]
        :type k: int
        :rtype: int
        """
        l,r=0,len(arr)-1
        while r>=l:
            mid=(l+r)//2
            missing_terms=arr[mid]-mid-1
           
            if missing_terms<k:
                l=mid+1
            else:
                r=mid-1
        return r+k+1
        

File: 1646-KthMissingPositiveNumber/1646-KthMissingPositiveNumber.cpp
// Last updated: 1/26/2026, 8:38:37 AM
class Solution {
public:
    int findKthPositive(vector<int>& arr, int k) {
        int low=0;
        int high=arr.size()-1;
        int mid;
        while(low<=high){
            mid=(low+high)/2;
            int mk=arr[mid]-(mid+1);
            if(mk>=k){
                high=mid-1;
            }else{
                low=mid+1;
            }
        }
        return k+high+1;
    }
};